frontend/ui/src/pages/Inbox/MessageInput/index.tsx

```diff
@@ -97,6 +101,38 @@ const MessageInput = (props: MessageInputProps & ConnectedProps<typeof connector
     setDisconnectedChannelToolTip(!conversation.channel.connected);
   }, [conversation.channel.connected]);
 
+  useEffect(() => {
+    if (selectedTemplate && templateSelectorDiv && templateSelectorDiv.current.offsetHeight > 200) {
+      const templateResizedHeight = 200;
+      const templateSelectorDivHeight = templateSelectorDiv.current.offsetHeight;
+      let iconSize;
+      let buttonSize;
+
+      const scaleRatio = Math.min(templateResizedHeight / templateSelectorDivHeight);
+
+      if (scaleRatio <= 0.7) {
+        if (scaleRatio > 0.3) {
+          iconSize = '18px';
+          buttonSize = '36px';
+        } else {
+          iconSize = '30px';
+          buttonSize = '60px';
+        }
+
+        setCloseIconHeight(iconSize);
+        setCloseIconWidth(iconSize);
+
+        if (removeTemplateButton && removeTemplateButton.current) {
+          removeTemplateButton.current.style.width = buttonSize;
+          removeTemplateButton.current.style.height = buttonSize;
+        }
+      }
+
+      templateSelectorDiv.current.style.transform = `scale(${scaleRatio})`;
+      templateSelectorDiv.current.style.transformOrigin = 'left';
+    }
+  }, [selectedTemplate]);
+
```

Comment:
- All the templates have different heights so we dynamically scale down the templates according to their original height.
- Once we've calculated the scale ratio, we scale down the remove button accordingly.